File: js-automation/pages/CheckoutPage.js
import { By } from 'selenium-webdriver';
import { BasePage } from './BasePage.js';

export class CheckoutPage extends BasePage {
  constructor(driver, baseUrl) {
    super(driver, baseUrl);
    this.fullNameInput = By.css("[data-testid='checkout-fullname']");
    this.emailInput = By.css("[data-testid='checkout-email']");
    this.phoneInput = By.css("[data-testid='checkout-phone']");
    this.streetInput = By.css("[data-testid='checkout-street']");
    this.cityInput = By.css("[data-testid='checkout-city']");
    this.stateInput = By.css("[data-testid='checkout-state']");
    this.postalCodeInput = By.css("[data-testid='checkout-postalcode']");
    this.placeOrderButton = By.css("[data-testid='place-order'], #place-order-submit-btn");
    this.orderSuccessContainer = By.css("[data-testid='order-success'], #order-success-container");
    this.orderErrorAlert = By.css("[data-testid='checkout-error']");
  }

  async load() {
    await this.navigateTo('/checkout');
    await this.waitForVisible(this.fullNameInput);
  }

  async fillShippingForm(data) {
    if (data.fullName) await this.type(this.fullNameInput, data.fullName);
    if (data.email) await this.type(this.emailInput, data.email);
    if (data.phone) await this.type(this.phoneInput, data.phone);
    if (data.street) await this.type(this.streetInput, data.street);
    if (data.city) await this.type(this.cityInput, data.city);
    if (data.state) await this.type(this.stateInput, data.state);
    if (data.postalCode) await this.type(this.postalCodeInput, data.postalCode);
  }

  async placeOrder() {
    await this.click(this.placeOrderButton);
    await this.sleep(1000);
  }

  async isOrderSuccessful() {
    return await this.isElementPresent(this.orderSuccessContainer, 12000);
  }
}

File: automation/Pages/CheckoutPage.cs
using System;
using OpenQA.Selenium;
using NOVAMart.Automation.Configuration;
using NOVAMart.Automation.Models;
using NOVAMart.Automation.Utilities;

namespace NOVAMart.Automation.Pages
{
    public class CheckoutPage : BasePage
    {
        public CheckoutPage(IWebDriver driver) : base(driver) { }

        public IWebElement FullNameInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-fullname']"));

        public IWebElement EmailInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-email']"));

        public IWebElement PhoneInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-phone']"));

        public IWebElement StreetInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-street']"));

        public IWebElement CityInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-city']"));

        public IWebElement StateInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-state']"));

        public IWebElement PostalCodeInput =>
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-postalcode']"));

        public IWebElement PlaceOrderButton =>
            WaitHelper.WaitForElementClickable(Driver, By.CssSelector("[data-testid='place-order']"));

        public void NavigateTo()
        {
            Driver.Navigate().GoToUrl($"{TestConfiguration.Instance.BaseUrl}/checkout");
            WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='place-order']"));
        }

        public void FillShippingDetails(CheckoutModel data)
        {
            FullNameInput.Clear();
            FullNameInput.SendKeys(data.FullName);

            EmailInput.Clear();
            EmailInput.SendKeys(data.Email);

            PhoneInput.Clear();
            PhoneInput.SendKeys(data.Phone);

            StreetInput.Clear();
            StreetInput.SendKeys(data.Address);

            CityInput.Clear();
            CityInput.SendKeys(data.City);

            StateInput.Clear();
            StateInput.SendKeys(data.State);

            PostalCodeInput.Clear();
            PostalCodeInput.SendKeys(data.PostalCode);
        }

        public void SelectPaymentMethod(string method)
        {
            var selector = method.ToLowerInvariant() switch
            {
                "upi" => "button:has(svg.text-cyan-600), button:contains('UPI')",
                "cod" => "button:has(svg.text-amber-600), button:contains('COD')",
                _ => "button:has(svg.text-indigo-600), button:contains('Card')"
            };

            try
            {
                var buttons = Driver.FindElements(By.CssSelector("button"));
                foreach (var btn in buttons)
                {
                    if (btn.Text.Contains(method, StringComparison.OrdinalIgnoreCase))
                    {
                        btn.Click();
                        break;
                    }
                }
            }
            catch (Exception)
            {
                // Default is card
            }
        }

        public void PlaceOrder()
        {
            PlaceOrderButton.Click();
            WaitHelper.WaitForUrlContains(Driver, "/order-confirmation", 15);
        }

        public bool IsErrorDisplayed()
        {
            try
            {
                var error = WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='checkout-error']"), 4);
                return error != null && error.Displayed;
            }
            catch
            {
                return false;
            }
        }
    }
}
